### PR TITLE
fix: Fix query shard service release before channel release

### PR DESCRIPTION
**Pull request by weiliu1031**
issue: #28405

remove the reference counter in query shard service, to avoid all kind of release query shard service error

### CODE REVIEW

File: internal/querynode/query_shard_service.go [hunk 1]
```diff
@@ -102,31 +101,38 @@ func (q *queryShardService) addQueryShard(collectionID UniqueID, channel Channel
...
 		return err
 	}
-	qs.inUse.Add(delta)
 	q.queryShards[channel] = qs
 	log.Info("Successfully add new query shard")
 	return nil
 }
 
-func (q *queryShardService) removeQueryShard(channel Channel, delta int64) error {
+func (q *queryShardService) removeQueryShard(channel Channel) error {
```
Comment (yah01) [inline internal/querynode/query_shard_service.go:109]: NIT: name to `tryRemoveQueryShard` would better to match what the method does

**Review by yah01 — COMMENTED** [review]
/lgtm

File: internal/querynode/query_shard_service.go
[the remark this answers is not in the document]
Comment (weiliu1031) [reply internal/querynode/query_shard_service.go:1]: query shard service will be removed only when all segments and channel has been released:

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**